core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -730,20 +733,46 @@ public AvroCompatibilityLevel getCompatibilityLevel(String subject)
   @Override
   public boolean isCompatible(String subject,
                               String newSchemaObj,
-                              String latestSchema)
+                              String latestSchema) 
       throws SchemaRegistryException {
-    AvroSchema newAvroSchema = AvroUtils.parseSchema(newSchemaObj);
-    AvroSchema latestAvroSchema = AvroUtils.parseSchema(latestSchema);
     if (latestSchema == null) {
       throw new InvalidSchemaException(
-          "Existing schema " + latestSchema + " is not a valid Avro schema");
+        "Latest schema not provided");
     }
+    return isCompatible(subject, newSchemaObj, Collections.singletonList(latestSchema));
+  }
+
+  /**
+   * @param previousSchemas Full schema history in chronological order
+   */
+  @Override
+  public boolean isCompatible(String subject,
+                              String newSchemaObj,
+                              List<String> previousSchemas)
+      throws SchemaRegistryException {
+    AvroSchema newAvroSchema = AvroUtils.parseSchema(newSchemaObj);
+    
+    if (previousSchemas == null || previousSchemas.isEmpty()) {
+      throw new InvalidSchemaException(
+          "Previous schema not provided");
+    }
+    
+    List<org.apache.avro.Schema> previousAvroSchemas = new ArrayList<>(previousSchemas.size());
+    for (String previousSchema : previousSchemas) {
```

Comment:
Not that it is horribly expensive to parse them, but should we consider rearranging (or adding to) how we store the schemas so we don't have to re-parse everything every time we check compatibility?